addons/account/models/partner.py

```diff
@@ -171,7 +171,7 @@ def get_fiscal_position(self, partner_id, delivery_id=None):
         delivery = PartnerObj.browse(delivery_id)
 
         # If partner and delivery have the same vat prefix, use invoicing
-        if not delivery.vat or (partner.vat and delivery.vat[:2] == partner.vat[:2]):
+        if not delivery or (delivery.vat and partner.vat and delivery.vat[:2] == partner.vat[:2]):
```

Comment:
Nearly. We should test if delivery is set first.

Reply:
@rousseldenis what do you mean with this comment ?

Reply:
towards first @william-andre suggestion (see above)